src/Validation/Results.php

```diff
@@ -0,0 +1,54 @@
+<?php
+/**
+ * This file is part of Lcobucci\JWT, a simple library to handle JWT and JWS
+ *
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD-3-Clause
+ */
+
+declare(strict_types=1);
+
+namespace Lcobucci\JWT\Validation;
+
+/**
+ * Results returned by validation
+ */
+class Results implements ResultsInterface
+{
+    /**
+     * @var array
+     */
+    private $errors = [];
+
+    /**
+     * @param string $name
+     * @param string $message
+     */
+    public function addError(string $name, string $message)
```

Comment:
I think you should inject the exception here instead of defining these two arguments, you can also specialize the exception class.